dev/NavigationView/NavigationView.cpp

```diff
@@ -398,7 +398,7 @@ void NavigationView::CreateAndHookEventsToSettings(std::wstring_view settingsNam
 
         if (shouldSelectSetting)
         { 
-            SetSelectedItemAndExpectItemInvokeWhenSelectionChangedIfNotInvokedFromAPI(nullptr);
+            SetSelectedItemAndExpectItemInvokeWhenSelectionChangedIfNotInvokedFromAPI(selectedItem);
```

Comment:
Could you please compare the ItemInvoke and SelectionChanged events with/without this change when you change DisplayMode?
We don't expect that any event is raised.

Reply:
Oh okay. So the expected behavior is to not raise a selection changed event?

The problem I see with this, is that technically, the selection changes when we look at the visual tree. If I recall correctly, when we change from top to left or vice versa, we use different components (e.g. we actually have 2 settings buttons, one for top mode and one for left mode). 

So TECHNICALLY, the VisualTree item that is selected has changed, however from a logic perspective the selection has not changed. Another problem I could see is that we maybe could code relying on exactly that behavior.

I hope that makes sense.😅 

Of course I am happy to change the fix for #148 if no selection changed event should be raised. :)

Reply:
@YuliKl to double confirm. When the selected item is settings, and we change it from left nav to top nav, do you expect these two events? For the user, I may not want them. 

I don't remember we officially support the dynamic flip between top nav and left nav, although our test case have this. And do you know if there are customer really using it?

Reply:
Dynamically changing between top nav and left (minimal) nav is a behavior the Store app uses. (Although Store doesn't use NavigationView's Settings item)

From a user's perspective, changes in how the nav is laid out doesn't equate to a change in which nav item is selected. In fact, for the user, the selected item has not changed at all, only the position on screen where that item is being drawn has changed. So I don't think we should raise any selected/invoked events here.

Reply:
Ah okay, I will fix that tomorrow.

Reply:
As I know, Store app uses two NavigationView to mimic it other than one.

Reply:
You would know better than I. I admit that I've never actually seen their code. I know the intent of their experience but not the implementation details.

Reply:
We do have a sample in the Gallery that shows how to switch between top and left nav using VSM triggers. So other apps may be choosing to do this.

Reply:
Fixed with latest commit. Now `SelectionChanged` does not get raised when we change the DisplayMode.